io/io/src/TBufferMergerFile.cxx

```diff
@@ -29,7 +29,24 @@ TBufferMergerFile::~TBufferMergerFile()
 
 Int_t TBufferMergerFile::Write(const char *name, Int_t opt, Int_t bufsize)
 {
+   // Make sure the compression of the basket is done in the unlocked thread and
+   // not in the locked section.
+   if (!fMerger.GetNotrees())
+      TMemFile::Write(name, opt | TObject::kOnlyPrepStep, bufsize);
+
+   // Instead of Writing the TTree, doing a memcpy, Pushing to the queue
+   // then Reading and then deleting, let's see if we can just merge using
+   // the live TTree.
+   if (fMerger.TryMerge(this)) {
+      ResetAfterMerge(0);
+      return 0;
+   }
+
+   auto oldCompLevel = GetCompressionLevel();
+   if (fMerger.GetCompressTemporaryKeys())
+      SetCompressionLevel(0);
```

Comment:
Isn't this backwards? You probably want to compress if `fMerger.GetCompressTemporaryKeys()` is true.

Reply:
indeed :(